### DISCUSSION TITLE
multi-sample velocity

### DISCUSSION THREAD

**Discussion by chansigit** [theislab/scvelo#877] [Q&A]
We sequenced multiple samples with similar biological states and individually performed scvelo analysis well.

I hope to merge these cells into an integrated anndata object and derive an overall velocity. However I am worried about the following issues:

1) does velocity matrix (u,s) have the same batch effect issues like the expression matrix? In expression analysis, the same groups of cells of two batches may blend poorly. If that happens in velocity analysis, the similar cells are no longer consecutive and may ruin the analysis.

2) the estimated velocity arrows sometimes suffer random biases. Is it possible that arrows in a local domain on a umap conflict with each other and cancel out due to the random discrepancy?

That's my worries, are there any scvelo solution coordinating multiple samples' velocities ?

Thank you!

**Comment by WeilerP** [comment]
@chansigit, simply merging the final AnnData objects of each sample doesn't make sense, IMO. If you want to project them in a common latent embedding (_e.g._ UMAP), for example, you will have to recalculate a new neighbor graph on the entire dataset which is independent of the one used for the individual samples. The one for the individual ones is used for calculating the velocities, _i.e._, there is no justification for using the neighbor graph on the entire dataset to project velocities.

> does velocity matrix (u,s) have the same batch effect issues like the expression matrix? 
Do you mean the velocity matrix or the matrices of unspliced and spliced counts? In both cases, I'd say yes since they come from the original expression matrix. If the same cell type in two different clusters differs due to batch effects, so will the velocities most likely. You can check by e.g. merging the velocities of the two samples and analysing a latent embedding.

> the estimated velocity arrows sometimes suffer random biases. Is it possible that arrows in a local domain on a umap conflict with each other and cancel out due to the random discrepancy?
Sorry, not sure I follow. Velocities are not aggregated over neighbors. For the embedding, a weighted, corrected average over the displacement between a reference cell and its neighbors. Consequently, the directions can cancel each other (for example in the case of two neighbors in opposite directions of the reference cell and equal transition probability).

**Reply by meghamsh738** [reply]
I have looked at a paper  where they have integrated data and projected velocities by combining loom files (data set had 4 different conditions, not replicates). I can either do this or integrate data, extract embeddings per condition and use individual condition's loom files for running velocity. When I tried both the ways, there were some differences in the arrows but not major. Which pipeline is advisable ?

**Reply by chansigit** [reply]
> 

can you please show me the paper link?

**Reply by meghamsh738** [reply]
https://pubmed.ncbi.nlm.nih.gov/35233577/

**Reply by chansigit** [reply]
> 

thank you :D

**Reply by meghamsh738** [reply]
Dear chansigit, I'm curious to know which method you have followed for the pipeline. Can you describe your steps briefly ? 
Thanks
